modules/core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManagerComponent.java

```diff
@@ -182,6 +182,14 @@ protected BlobProvider getBlobProvider(String key, String repositoryName) {
      */
     @Override
     public String writeBlob(Blob blob, Document doc, String xpath) throws IOException {
+        if (blob == null) {
+            if (MAIN_BLOB_XPATH.equals(xpath) && doc.isUnderRetentionOrLegalHold()) {
+                throw new DocumentSecurityException(
+                        "Cannot delete blob from document " + doc.getUUID() + ", it is under retention / hold");
+            }
+            return null;
+        }
```

Comment:
What do you think about adding a new API such as `isBlobWriteable` which does this check, and maybe the one l. 211(?), in order to avoid to allow null values for _blob_ in this API?

Reply:
Not sure if I understood `and maybe the one l. 211(?)`, could you please elaborate a little bit?

Regarding the new API, would it also make sense for the 10.10 backport?
Thanks!

Reply:
>Not sure if I understood and maybe the one l. 211(?), could you please elaborate a little bit?

Sure, I was talking about this part of the code https://github.com/nuxeo/nuxeo/blob/b65f2dd57707081d3f7451f90dc96ea01e8df7c5/modules/core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManagerComponent.java#L219-L222 which could leverage
the new API.

>Regarding the new API, would it also make sense for the 10.10 backport?

It would make sense for the 10.10 backport if the API has a default implementation.

Reply:
It's not clear to me on what class you want to add this new API.
As it's a detail, let's postpone refactorings and just do a minimal change + backport.